stores/metadata.go

```diff
@@ -1687,6 +1689,7 @@ func (s *SQLStore) DeleteHostSector(ctx context.Context, hk types.PublicKey, roo
 		}
 		return nil
 	})
+	return deletedSectors, err
```

Comment:
This is not entirely correct, we should `return 0, err` when the error is not `nil` because of the transaction. I know the caller would usually not even evaluate `deletedSectors` if the error is not `nil` but still